lib/numbers_and_words/strategies/figures_converter/languages/vi.rb

```diff
@@ -0,0 +1,96 @@
+module NumbersAndWords
+  module Strategies
+    module FiguresConverter
+      module Languages
+        class Vi < Base
+          def print_words
+            value = super
+            remove_billion_noice(value)
+          end
+
+          def strings_logic
+            if fractional?
+              if fraction_number_zeros_leading.zero?
+                @options.options[:integeral] = true
+                integeral_string_logic
+              else
+                fractional_with_zero_leading_string_logic
+              end
+            else
+              integeral_string_logic
+            end
+          end
+
+          def fractional_with_zero_leading_string_logic
+            leading_zeros = [0] * fraction_number_zeros_leading
+            (@figures + leading_zeros).map(&:to_words)
+          end
+
+          def integeral_string_logic
+            if @figures.capacity_count
+              number_without_capacity_to_words + complex_number_to_words
+            elsif hundreds?
+              hundreds_number_to_words
+            elsif @figures.tens || @figures.ones
+              simple_number_to_words
+            else
+              []
+            end
+          end
+
+          def hundreds?
+            @figures.hundreds ||
+              (@figures.size == 3 && (@figures.teens || @figures.tens || @figures.ones))
+          end
+
+          def hundreds(options = {})
+            if @figures.hundreds.to_i.zero?
+              translate :hundreds, 0, options
```

Comment:
Coveralls is complaining that this line is not getting tested.

Reply:
I cover it in commit *ff7d3fac9990536aca08207f931bdc0749f5c1ff*